src/v2i-hub/SpatPlugin/src/SignalControllerConnection.cpp

```diff
@@ -37,7 +37,7 @@ namespace SpatPlugin {
         return status;
     };
 
-    void SignalControllerConnection::receiveBinarySPAT(const std::shared_ptr<SPAT> &spat, uint64_t timeMs ) const {
+    void SignalControllerConnection::receiveBinarySPAT(SPAT *spat, uint64_t timeMs ) const {
```

Comment:
I see the prior ptr was const, should this be `const SPAT *...`?

Reply:
Before it was a const reference to a shared pointer. This means that the method did not increase the reference count for the smart pointer. This is done when passing a smart pointer to a function which should non outlive the calling function. Essentially this is sharing a smart pointer without passing ownership of the smart pointer to a function.  

This pointer is modified by this function so I don't think I can  make it const.

Reply:
Corrected to this syntax ` SPAT *  const spat` which gives you a const pointer where the pointer can not be modified to point to a difference address but the value can be modified (https://www.geeksforgeeks.org/c/difference-between-const-int-const-int-const-and-int-const/).